lib/api/apiUtils/object/coldStorage.js

```diff
@@ -37,21 +35,17 @@ function getAmzRestoreResHeader(objMD) {
 /**
  * Check if restore can be done.
  *
- * @param {BucketInfo} bucketMD - bucket instance
  * @param {ObjectMD} objectMD - object metadata
  * @param {object} log - werelogs logger
  * @return {ArsenalError|boolean} - undefined if the conditions for RestoreObject are fulfilled
  */
-function _validateStartRestore(bucketMD, objectMD, log) {
-    const isLocationCold = locationConstraints[bucketMD.getLocationConstraint()]?.isCold
-        || locationConstraints[objectMD.dataStoreName]?.isCold;
+function _validateStartRestore(objectMD, log) {
+    const isLocationCold = !!objectMD.archive;
```

Comment:
hi @nicolas2bert  @naren-rajendran, 
we changed the way to check if the object is stored in cold storage or hot. 
Since bucketMD.locationConstraint might not change when we archive an object of the bucket, I removed the line checking bucket locationConstraint.
we use `archive` field of objectMD to check, we assume if this field exists, then the object should have been archive.
Could you please confirm with us that if this check look good?